ledger/acctupdates.go

```diff
@@ -739,6 +741,18 @@ func (au *accountUpdates) Totals(rnd basics.Round) (totals AccountTotals, err er
 	return au.totalsImpl(rnd)
 }
 
+func (au *accountUpdates) CountStorageForRound(rnd basics.Round, addr basics.Address, aidx basics.AppIndex, global bool) (basics.StateSchema, error) {
+	au.accountsMu.RLock()
+	defer au.accountsMu.RUnlock()
+	return au.countStorageForRound(rnd, addr, aidx, global)
```

Comment:
while it won't be incorrect, it could generate errors due to database writing orders. To avoid that, make sure you pass syncronized=true to the underlying lookupWithoutRewards method, instead of taking the lock yourself.

Reply:
(admittedly, I'm not sure if this function even needs to be here; it could be implemented by a higher level completely )